client/shaders/wielded_shader/opengl_vertex.glsl

```diff
@@ -23,10 +28,36 @@ void main(void)
 	vPosition = gl_Position.xyz;
 	worldPosition = (mWorld * gl_Vertex).xyz;
 
-	vec3 sunPosition = vec3 (0.0, eyePosition.y * BS + 900.0, 0.0);
-
-	lightVec = sunPosition - worldPosition;
 	eyeVec = -(gl_ModelViewMatrix * gl_Vertex).xyz;
 
-	gl_FrontColor = gl_BackColor = gl_Color;
+	// Calculate color.
+	// Red, green and blue components are pre-multiplied with
+	// the brightness, so now we have to multiply these
+	// colors with the color of the incoming light.
+	// The pre-baked colors are halved to prevent overflow.
+	vec4 color = gl_Color;
+	color.a = 1.0;
+
+	// The alpha gives the ratio of sunlight in the incoming light.
+	float nightRatio = 1.0 - gl_Color.a;
+
+#ifdef ENABLE_ADVANCED_LIGHTING
+	vec3 norm = normalize((mWorld * vec4(gl_Normal, 0.0)).xyz);
+
+	// Directional shading color
+	vec3 resultLightColor = ((lightColor.rgb * gl_Color.a) + nightRatio) *
+		((max(dot(norm, lightDirection), -0.2) + 0.2) / 1.2) * lightColor.a;
+
+	resultLightColor = (resultLightColor * 0.6) + 0.4;
+
+	color.rgb *= resultLightColor;
+#endif
+
+        // Emphase blue a bit in darker places
+        // See C++ implementation in mapblock_mesh.cpp finalColorBlend()
+        float brightness = (color.r + color.g + color.b) / 3;
+	color.b += max(0.0, 0.021 - abs(0.2 * brightness - 0.021) +
```

Comment:
Hm, are you using 7 space tabs? I indented with a tab, the original code seems to use 8 spaces.